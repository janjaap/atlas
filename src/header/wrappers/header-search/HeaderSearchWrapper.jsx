import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'react-dom';

import SearchContainer from '../../containers/header-search/HeaderSearchContainer';

const HeaderSearchWrapper = (props) => (
  <Provider store={window.reduxStore}>
    <SearchContainer {...props} />
  </Provider>
);

export default HeaderSearchWrapper;

window.React = window.React || React;
window.render = window.render || render;
window.SearchWrapper = HeaderSearchWrapper;
