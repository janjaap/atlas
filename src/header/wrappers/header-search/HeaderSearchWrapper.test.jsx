import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme/build/index';
import HeaderSearchWrapper from './HeaderSearchWrapper';

describe('HeaderSearchWrapper', () => {
  const initialState = {
    autoSuggest: {
      count: 6,
      suggestions: [
        {
          content: [
            {
              uri: 'bag/openbareruimte/03630000003186/',
              label: 'Dam',
              index: 0,
              category: 'Straatnamen'
            },
            {
              uri: 'bag/openbareruimte/03630000001038/',
              label: 'Damloperspad',
              index: 1,
              category: 'Straatnamen'
            },
            {
              uri: 'bag/openbareruimte/03630000003187/',
              label: 'Damrak',
              index: 2,
              category: 'Straatnamen'
            }
          ],
          label: 'Straatnamen'
        },
        {
          content: [
            {
              uri: 'monumenten/monumenten/8e8ae1dd-78a2-4a5f-841d-1870631b7e33/',
              label: 'Damrak 1',
              index: 3,
              category: 'Monumenten'
            },
            {
              uri: 'monumenten/monumenten/aa3f9081-2ac4-49ea-95d2-0aad7aecd883/',
              label: 'Dam 10',
              index: 4,
              category: 'Monumenten'
            },
            {
              uri: 'monumenten/monumenten/f93e31ba-89eb-4784-87e1-d32c33b5236d/',
              label: 'Damrak 15',
              index: 5,
              category: 'Monumenten'
            }
          ],
          label: 'Monumenten'
        }
      ],
      typedQuery: 'dam'
    }
  };

  it('should render', () => {
    const store = configureMockStore()({ ...initialState });
    global.window.reduxStore = store;
    jest.spyOn(store, 'dispatch');
    const component = shallow(<HeaderSearchWrapper />, { context: { store } }).dive().dive();

    expect(component).toMatchSnapshot();
  });
});
