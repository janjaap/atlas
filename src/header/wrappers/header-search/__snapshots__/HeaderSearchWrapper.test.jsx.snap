// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearchWrapper should render 1`] = `
<HeaderSearchContainer
  activeSuggestion={
    Object {
      "index": -1,
    }
  }
  displayQuery=""
  isDatasetView={false}
  isMapFullscreen={false}
  numberOfSuggestions={6}
  onCleanDatasetOverview={[Function]}
  onDatasetSearch={[Function]}
  onDetailLoad={[Function]}
  onGetSuggestions={[Function]}
  onSearch={[Function]}
  onSuggestionActivate={[Function]}
  pageName=""
  prefillQuery=""
  suggestions={
    Array [
      Object {
        "content": Array [
          Object {
            "category": "Straatnamen",
            "index": 0,
            "label": "Dam",
            "uri": "bag/openbareruimte/03630000003186/",
          },
          Object {
            "category": "Straatnamen",
            "index": 1,
            "label": "Damloperspad",
            "uri": "bag/openbareruimte/03630000001038/",
          },
          Object {
            "category": "Straatnamen",
            "index": 2,
            "label": "Damrak",
            "uri": "bag/openbareruimte/03630000003187/",
          },
        ],
        "label": "Straatnamen",
      },
      Object {
        "content": Array [
          Object {
            "category": "Monumenten",
            "index": 3,
            "label": "Damrak 1",
            "uri": "monumenten/monumenten/8e8ae1dd-78a2-4a5f-841d-1870631b7e33/",
          },
          Object {
            "category": "Monumenten",
            "index": 4,
            "label": "Dam 10",
            "uri": "monumenten/monumenten/aa3f9081-2ac4-49ea-95d2-0aad7aecd883/",
          },
          Object {
            "category": "Monumenten",
            "index": 5,
            "label": "Damrak 15",
            "uri": "monumenten/monumenten/f93e31ba-89eb-4784-87e1-d32c33b5236d/",
          },
        ],
        "label": "Monumenten",
      },
    ]
  }
  typedQuery="dam"
/>
`;
